build-tools/packages/build-cli/src/codeCoverageAnalysisTool/getCommentForCodeCoverage.ts

```diff
@@ -0,0 +1,159 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { IBaselineBuildMetrics } from "./ADO/getBaselineBuildMetrics.js";
+import type { CodeCoverageSummary } from "./codeCoverageCli.js";
+import type { CodeCoverageComparison } from "./compareCodeCoverage.js";
+
+const codeCoverageDetailsHeader = `<table><tr><th>Metric Name</th><th>Baseline coverage</th><th>PR coverage</th><th>Coverage Diff</th></tr>`;
+
+/**
+ * Method that returns the comment to be posted on PRs about code coverage
+ * @param codeCoverageComparisonReport - The comparison report between baseline and pr test coverage
+ * @param adoPrId - Pull Request ID for the current PR
+ * @param adoConnection - A connection to the ADO api.
+ * @returns Comment to be posted on the PR
+ */
+export const getCommentForCodeCoverageDiff = async (
+	codeCoverageComparisonReport: CodeCoverageComparison[],
+	baselineBuildInfo: IBaselineBuildMetrics,
+): Promise<CodeCoverageSummary> => {
+	// Find new packages that do not have test setup and are being impacted by changes in the PR
+	const newPackagesIdentifiedByCodeCoverage = codeCoverageComparisonReport.filter(
+		(codeCoverageReport) =>
+			(codeCoverageReport.lineCoverageInPr === 0 &&
+				codeCoverageReport.lineCoverageInBaseline === 0) ||
+			codeCoverageReport.isNewPackage,
+	);
+	console.log(`Found ${newPackagesIdentifiedByCodeCoverage.length} new packages`);
+
+	// Find existing packages that have reported a change in coverage for the current PR
+	const existingPackagesWithCoverageChange = codeCoverageComparisonReport.filter(
+		(codeCoverageReport) =>
+			codeCoverageReport.branchCoverageDiff !== 0 || codeCoverageReport.lineCoverageDiff !== 0,
+	);
+	console.log(
+		`Found ${existingPackagesWithCoverageChange.length} packages with code coverage changes`,
+	);
+
+	const packagesWithNotableRegressions = existingPackagesWithCoverageChange.filter(
+		(codeCoverageReport: CodeCoverageComparison) =>
+			codeCoverageReport.branchCoverageDiff < -1 || codeCoverageReport.lineCoverageDiff < -1,
```

Comment:
We confirmed the percentages are reported as `77.81` and not `0.7781`, right? So the comparisons in this file don't need to be adjusted to a 0-1 range?